vortex-dtype/src/struct_.rs

```diff
@@ -346,7 +350,17 @@ impl StructFields {
             .map(|(_, dtype)| dtype.clone())
             .collect::<Vec<_>>();
 
-        StructFields::from_fields(names, dtypes)
+        Ok(StructFields::from_fields(names, dtypes))
+    }
+
+    /// Returns a new [`StructFields`] without the field at the given index.
+    ///
+    /// ## Panics
+    /// Panics if the index is out of bounds for the struct fields.
+    /// Prefer using `try_without_field` for fallible removal.
+    pub fn without_field(&self, index: usize) -> Self {
+        self.try_without_field(index)
+            .unwrap_or_else(|e| vortex_panic!("Failed to remove field: {}", e))
```

Comment:
how is this better than having one method that just panics? Where would you use this instead of the result returning method?

Reply:
better to have the fallible version instead of panicking, because the main caller wants to return `None` if the field doesn't exist, but will currently panic